pkg/apis/servicecatalog/types.go

```diff
@@ -537,8 +537,20 @@ type ServiceInstanceStatus struct {
 	// ReconciledGeneration is the 'Generation' of the serviceInstanceSpec that
 	// was last processed by the controller. The reconciled generation is updated
 	// even if the controller failed to process the spec.
+	// Deprecated: use ObservedGeneration instead
 	ReconciledGeneration int64
 
+	// ObservedGeneration is the 'Generation' of the serviceInstanceSpec that
+	// was last processed by the controller. The observed generation is updated
+	// whenever the status is updated regardless of operation result
+	ObservedGeneration int64
+
+	// Provisioned is the flag that marks whether the instance has been
+	// successfully provisioned (i.e. it exists on the broker's side and can be
+	// updated or deprovisioned). The flag should be reset after successful
+	// deprovisioning.
+	Provisioned bool
```

Comment:
`Provisioned` seems like something appropriate for a condition as opposed to bool.

Reply:
@kibbles-n-bytes Yeah I agree, that might be a good idea.
Not sure if we'll need to distinguish the initial "unprovisioned" state from the "deprovisioned" (after deletion or orphan mitigation) there.

Reply:
@nilebox @kibbles-n-bytes I disagree that provisioned should be a condition as opposed to a bool. The move in kube is to use simple fields instead of conditions. See https://github.com/kubernetes/kubernetes/issues/7856#issuecomment-335687733.

Reply:
Yes, we should avoid using bool fields. How about `ProvisionedStatus` for the field and an enum of `{NotProvisioned,Provisioned}` for the values it can take on, similar to the `Deprovision` status field?

Reply:
@staebler I don't really have a preference, but the quorum seems to advocate for `ProvisionedStatus`...

Reply:
@nilebox That's fine. My issue was using a condition (in the traditional kube sense) as opposed to a field, where kube is moving towards preferring a field. I have no reservation with using `ProvisionedStatus` instead of a bool.